patroni/daemon.py

```diff
@@ -0,0 +1,73 @@
+import abc
+import os
+import signal
+import six
+import sys
+
+from threading import Lock
+
+
+@six.add_metaclass(abc.ABCMeta)
+class AbstractPatroniDaemon(object):
+
+    def __init__(self):
+        from patroni.config import Config
+        from patroni.log import PatroniLogger
+
+        self.setup_signal_handlers()
+
+        self.logger = PatroniLogger()
+        self.config = Config()
+        self.logger.reload_config(self.config.get('log', {}))
+
+    def sighup_handler(self, *args):
+        self._received_sighup = True
+
+    def sigterm_handler(self, *args):
+        with self._sigterm_lock:
+            if not self._received_sigterm:
```

Comment:
just out of curiosity - does this `if` protect against repetitive `SIGTERM`s while the `SystemExit` is being processed ?